### DISCUSSION TITLE
Add functionality to programmatically update the route of a window

### DISCUSSION THREAD

**Discussion by Tray2** [NativePHP/laravel#358] [Ideas]
When using the application menu, you  should be able to programmatically change the route for a given window.

Something like this would be really clean.

```php
Menu::new()
  ->link(route('books.create'), 'main')
```

The first parameter is the route you want to go to, and the second the window that you want to do it in.
Maybe even default to `main` when a route is given.

That way you don't need to register an event and listen for it.

However if you want to do that, then it would be nice if you could go something like

```php
Window:get('main')
  ->goTo(route('books.create'));
```

Same as when you do

```php
Window::current()
  ->goTo(route('books.create'));
```

**Comment by simonhamp** [comment]
Yeh, I agree, this would be really nice. I'm currently refactoring the Menu API as it's a little clunky, so I'm going to add this into it

**Comment by simonhamp** [comment]
You can [now](https://github.com/NativePHP/laravel/pull/391) use the `Window::url()` method to update a window's URL.

**Reply by Tray2** [reply]
Sweet, will give that a go asap.

**Comment by justrau** [comment]
Managed to make it work using Window::url(), thanks, Simon!

---

Full example in case someone else needs this as well:
`App\Providers\NativeAppServiceProvider@boot`
```php
Menu::new()
    ->appMenu()
    ->fileMenu()
    ->editMenu()
    ->viewMenu()
    ->windowMenu()
    ->submenu('Tools', Menu::new()
        ->event(\App\Events\Menu\RestoreBackupMenuItemClicked::class, 'Restore Backup')
    )
    ->register();
```

`\App\Events\Menu\RestoreBackupMenuItemClicked`
```php
<?php

namespace App\Events\Menu;

class RestoreBackupMenuItemClicked
{
}
```

`App\Listeners\Menu\RedirectToRestoreBackupPage`
```php
<?php

namespace App\Listeners\Menu;

use App\Events\Menu\RestoreBackupMenuItemClicked;
use Window;

class RedirectToRestoreBackupPage
{
    public function handle(RestoreBackupMenuItemClicked $event): void
    {
        Window::get('main')->url(route('plain.restore-backup'));
    }
}
```

**Reply by simonhamp** [reply]
Nice. We'll definitely make this easier in a future release

**Comment by simonhamp** [comment]
Here's the PR that will allow us to do this [directly from a menu item](https://github.com/NativePHP/laravel/pull/423). It'll look something like this:

```php
Menu::goToUrl(url('/path/to/page'));

// or

Menu::goToRoute('route.name')
```

**Comment by simonhamp** [comment]
This was [released recently](https://nativephp.com/docs/1/the-basics/application-menu#link-items)
